src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/CngAsnFormatter.cs

```diff
@@ -17,9 +18,38 @@ protected override string FormatNative(Oid oid, byte[] rawData, bool multiLine)
             // to use hex formatting by providing an empty OID string.
             string oidValue = string.Empty;
             if (oid != null && oid.Value != null)
+            {
                 oidValue = oid.Value;
+            }
 
-            return Cng.CryptFormatObject(oidValue, rawData, multiLine);
+            int dwFormatStrType = multiLine ? Interop.Crypt32.CRYPT_FORMAT_STR_MULTI_LINE : Interop.Crypt32.CRYPT_FORMAT_STR_NONE;
+            int cbFormat = 0;
+            const int X509_ASN_ENCODING = 0x00000001;
+            unsafe
+            {
+                IntPtr oidValuePtr = Marshal.StringToHGlobalAnsi(oidValue);
+                try
+                {
+                    if (Interop.Crypt32.CryptFormatObject(X509_ASN_ENCODING, 0, dwFormatStrType, IntPtr.Zero, (byte*)oidValuePtr, rawData, rawData.Length, null, ref cbFormat))
+                    {
+                        int charLength = (cbFormat + 1) / 2;
+                        Span<char> buffer = charLength <= 256 ? stackalloc char[256] : new char[charLength];
```

Comment:
No array pool?

Reply:
You tell me :)  I wasn't sure how a) common it would be to have a charLength > 256, nor b) how hot these paths might be.

Reply:
Thanks. Char buffers are common. I'll add the pooling.